http_check/datadog_checks/http_check/http_check.py

```diff
@@ -133,7 +133,7 @@ def send_status_down(loginfo, down_msg):
                 (self.SC_STATUS, AgentCheck.CRITICAL, "{}. Connection failed after {} ms".format(str(e), length))
             )
 
-        except socket.error as e:
+        except (socket.error, IOError) as e:  # Py2 throws IOError on invalid cert path while py3 throws a socket.error
```

Comment:
It seems that both Py2 and Py3 raise an IOError (to be precise an OSError in Py3, but OSError and IOError are now the same object).

I don't think we should reuse that except block the host is not DOWN, maybe we can add a new one to log something more useful?

Reply:
I'm not sure about this myself - `socket.error` derives from `OSError`/`IOError` so I don't think we can discriminate easily between which of the host side or the Agent side have an issue.

In either case I think it's okay to report the check as failing, but then maybe the message could be less definitive? Eg "The host _could not be reached_: ..." (which might be because it's down, OR because of a check configuration issue).

Reply:
The exception message will give the details about what's going on, for example:

```
IOError: Could not find a suitable TLS CA certificate bundle, invalid path: /test
```

I see two options here:

- we leave the socket.error block as before and add a new block after dedicated to catching the more generic IOError/OSError and logging the error message without logging "host is down"
- Or we just let things as they were before, knowing that the exception is caught in the next except block and logs the exception content as expected